Add OGC API Features support (#11535)
camptocamp/c2cgeoportal: doc/integrator/ogc_api.rst

.. _integrator_ogc_api:

OGC API Features
----------------

The OGC API is a new standard that defines modular API following REST standard to access spatial data.

> The current implementation is only on the server side and is based on the OGC API - Features standard and only works with the GetFeature request using a bbox filter.

Reviewer: ```suggestion
The current implementation is only on the server side and is based on the OGC API - Features standard and only works with the `items` request using a filter.
```

> It is currently supported by both MapServer and QGISserver.

Reviewer: ```suggestion
It is currently supported by both MapServer and QGIS Server.
```

The following variables should be added to the docker-compose-lib.yaml file for the 'MapServer' service:

.. prompt:: bash::

    environment:
        ...

>         - MAPSERVER_BASE_PATH=/mapserv_proxy/mapserver

Reviewer: ```suggestion
        - MAPSERVER_BASE_PATH=/mapserv_proxy
```

        - OGCAPI_HTML_TEMPLATE_DIRECTORY=/usr/local/share/mapserver/ogcapi/templates/html-bootstrap4/
        ...

> The following changes has to be done to the Mapfiles to support the OGC API:

Reviewer: ```suggestion
The following changes has to be done to the MapFiles to support the OGC API:
```

.. prompt:: bash::

    MAP
        ...
        WEB
            ...

>             TEMPLATE "dummy""

Reviewer: Do you verify that's relay needed, I don't think that needed.

            ...
        END
        ...
    END

> QGISserver is already configured to support the OGC API and detect automatically if wfs3 path is used.

Reviewer: ```suggestion
QGIS Server is already configured to support the OGC API and detect automatically if wfs3 path is used.
```

Landing pages for both MapServer and QGISserver are not supported yet.

OGC API features are accessible through mapserv_proxy, with the following URLs:

> * ``/mapserv_proxy/<ogc-server>/ogcapi/*``: The Mapserver path.

Reviewer: ```suggestion
* ``/mapserv_proxy/<ogc-server>/ogcapi/*``: The MapServer path.
```

> * ``/mapserv_proxy/<ogc-server>/wfs3/*``: The QGISserver path.

Reviewer: ```suggestion
* ``/mapserv_proxy/<ogc-server>/wfs3/*``: The QGIS Server path.
```

OGC API Documentation
---------------------

> Mapserver documentation:

Reviewer: ```suggestion
MapServer documentation:
```

https://mapserver.org/ogc/ogc_api.html

> QGISserver documentation:

Reviewer: ```suggestion
QGIS Server documentation:
```

> https://docs.qgis.org/3.34/en/docs/server_manual/services/ogcapif.html

Reviewer: Missing documentation:

- Documentation about MapServer map alias => update the URL in the related OGC Server.
- Remove all special characters in the OGC Server names because they appear in the par of the URL (recommended).

Author: Done.

.. _integrator_ogc_api:

OGC API Features
----------------

> The OGC API is a new standard that defines modular API following REST standard to access spatial data.

Reviewer 2: defines modular -> defines a modular
following REST -> following the REST

> The current implementation is only on the server side and is based on the OGC API - Features standard and only works with the `items` request using a bbox filter.

Reviewer 2: The current implementation of what? 
I assume we are describing what parts of the OCG API GeoMapFish implements?

It is currently supported by both MapServer and QGIS Server.

> The following new environment variables should be added to the docker-compose-lib.yaml file for the 'MapServer' service:

Reviewer 2: I assume we should do this if we want the MapServer contained in GeoMapFish to provide an OGC API service?

.. prompt:: bash::

    environment:
        ...
        - MAPSERVER_BASE_PATH=/mapserv_proxy

>         - OGCAPI_HTML_TEMPLATE_DIRECTORY=/usr/local/share/mapserver/ogcapi/templates/html-bootstrap4/

Reviewer: This should be added to the `docker-compose-lib.yaml` provided by c2cgeoportal!

Author: So I can ignore it in the doc ?

Reviewer: yes
